packages/mdc-base/foundation.ts

```diff
@@ -21,44 +21,35 @@
  * THE SOFTWARE.
  */
 
-/**
- * @template A
- */
-class MDCFoundation {
-  /** @return enum{cssClasses} */
-  static get cssClasses() {
+class MDCFoundation<AdapterType extends {} = {}> {
+  static get cssClasses(): {} {
```

Comment:
Should it be possible for us to type these more specifically using index signatures? i.e. `cssClasses` and `strings` are always `{[key: string]: string}`, and `numbers` is always `{[key: string]: number}`.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types-and-string-index-signatures